go/libraries/doltcore/sqle/enginetest/dolt_queries_diff.go

```diff
@@ -4862,6 +4862,99 @@ var CommitDiffSystemTableScriptTests = []queries.ScriptTest{
 	},
 }
 
+var SchemaDiffSystemTableScriptTests = []queries.ScriptTest{
+	{
+		Name: "basic schema changes",
+		SetUpScript: []string{
+			"create table test (pk int primary key, c1 int, c2 int);",
+			"call dolt_add('.')",
+			"call dolt_commit('-m', 'commit 1');",
+			"alter table test drop column c2, add column c3 varchar(10);",
+			"call dolt_add('.')",
+			"call dolt_commit('-m', 'commit 2');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
```

Comment:
There seems to be a high reliance on HEAD^ and HEAD in these tests. Adding more commits would be a good idea, and tests with commit ids and tags would be a good addition as well.

The reason I call this out is that there are default situations where we use HEAD^ and HEAD - by virtue of that being the top of the history. Having more changes, then calculating the schema changes between them would surface more problems. For example, what if you compare two commits which don't have any schema differences even though they are different commits?